packages/rich-list-core/src/lib/SingleIndexDb.ts

```diff
@@ -0,0 +1,72 @@
+export interface Schema<T> {
+  id: string
+  partition: string
+  sort: number
+  data: T
+}
+
+export interface FindOptions {
+  partition: string
+  order: 'ASC' | 'DESC'
+  limit: number
+  gt?: number
+  lt?: number
+}
+
+export interface SingleIndexDb<T> {
+  put: (data: Schema<T>) => Promise<void>
+  delete: (id: string) => Promise<void>
+  get: (id: string) => Promise<Schema<T> | undefined>
+  list: (filter: FindOptions) => Promise<Array<Schema<T>>>
+}
+
+/**
+ * For unit test use.
+ */
+export class InMemoryDatabase<T> implements SingleIndexDb<T> {
+  private data: { [key: string]: Array<Schema<T>> } = {}
```

Comment:
Thoughts about using the syntatic sugar `Record<string, Array<Schema<T>>>` for representing this? Curious about your ideas.

Reply:
Practically no difference, but I do agree using `Record` will at least have better readability.